### PR TITLE
Issue #1449 new

### PR DESCRIPTION
**Pull request by stefanhaerter**
Closing #1449

### CODE REVIEW

File: var/httpd/htdocs/js/Core.Agent.DynamicFieldDBSearch.js [hunk 1]
```diff
@@ -258,13 +258,16 @@ Core.Agent.DynamicFieldDBSearch = (function(TargetNS) {
                     /TicketID=(\d+)/.exec(document.URL);
                     TicketID = RegExp.$1;
 
+                    var liElementId = $Element.parents().closest('li').attr('id');
+                    var activityDialogId = liElementId.match(/^Process_ActivityDialog-([a-f0-9]{32})$/)[1];
```
Comment (svenoe) [inline var/httpd/htdocs/js/Core.Agent.DynamicFieldDBSearch.js]: We use capitalized first letters for variables, in js, too.

File: var/httpd/htdocs/js/Core.Agent.DynamicFieldDBSearch.js [hunk 2]
```diff
@@ -258,13 +258,16 @@ Core.Agent.DynamicFieldDBSearch = (function(TargetNS) {
                     /TicketID=(\d+)/.exec(document.URL);
                     TicketID = RegExp.$1;
 
+                    var liElementId = $Element.parents().closest('li').attr('id');
```
Comment (svenoe) [inline var/httpd/htdocs/js/Core.Agent.DynamicFieldDBSearch.js]: It is better to be a bit more precise with the selectors. In this case, e.g. you could use 'li.Activity'. This way you circumvent any strange sideeffects, for new dynamic fields containing lists, e.g.. (Ok, unlikely, but still.)

File: var/httpd/htdocs/js/Core.Agent.DynamicFieldDBSearch.js [hunk 3]
```diff
@@ -258,13 +258,16 @@ Core.Agent.DynamicFieldDBSearch = (function(TargetNS) {
...
+                    var activityDialogId = liElementId.match(/^Process_ActivityDialog-([a-f0-9]{32})$/)[1];
+
                     // serialize form
                     QueryString = Core.AJAX.SerializeForm($('#'+DynamicFieldName).closest('form'), IgnoreList) + SerializeData(UpdateList);
 
                     QueryString += ";Action="+FrontendInterface;
                     QueryString += ";Term="+encodeURIComponent(Request.term);
                     QueryString += ";MaxResults="+Core.Config.Get('Autocomplete.MaxResultsDisplayed');
-                    QueryString += ";DynamicFieldName="+encodeURIComponent(DynamicFieldName);
+                    QueryString += ";DynamicFieldName="+encodeURIComponent(DynamicFieldName.substring(0, DynamicFieldName.indexOf('_' + activityDialogId)));
```
Comment (svenoe) [inline var/httpd/htdocs/js/Core.Agent.DynamicFieldDBSearch.js]: Have you tested whether that works for DB fields which are not part of a process? E.g. in CustomerTicketMessage, or on the agent interface? I would assume a check whether your activityDialogId exists would be good, even if just a warning is thrown.

Comment (stefanhaerter) [reply var/httpd/htdocs/js/Core.Agent.DynamicFieldDBSearch.js]: Indeed, this caused an error. Should be fixed with the latest commit

File: var/httpd/htdocs/js/Core.Agent.DynamicFieldDBSearch.js [hunk 4]
```diff
@@ -258,13 +258,21 @@ Core.Agent.DynamicFieldDBSearch = (function(TargetNS) {
                     /TicketID=(\d+)/.exec(document.URL);
                     TicketID = RegExp.$1;
 
+                    // for usage in process context
+                    var LiElementId = $Element.parents().closest('li.Activity').attr('id');
+                    var DynamicFieldNameQuery = DynamicFieldName;
+                    if (typeof LiElementId !== 'undefined') {
```
Comment (svenoe) [inline var/httpd/htdocs/js/Core.Agent.DynamicFieldDBSearch.js]: Did you try that for the normal reply (id=FollowUp) in CustomerTicketZoom? It also has the Activity-class, but the substitution is not needed. Maybe it is necessary to include the "match" as "test" in the if condition.

File: var/httpd/htdocs/js/Core.Agent.DynamicFieldDBSearch.js [hunk 5]
```diff
@@ -276,7 +284,7 @@ Core.Agent.DynamicFieldDBSearch = (function(TargetNS) {
                         // run the response function to hide the request animation
                         Response({});
                     }
-
+
```
Comment (svenoe) [inline var/httpd/htdocs/js/Core.Agent.DynamicFieldDBSearch.js:290]: Please remove the blanks. Usually this is done by the CodePolicy, but we have to improve it, to run with js in a meaningful way, too...

**Comment by stefanhaerter** [comment]
Closing this and shifting changes to new branch

**Comment by stefanhaerter** [comment]
Unsetting the milestone as we do not use milestones on PRs.

**Pull request closed without merging**